yaml_test.go

```diff
@@ -0,0 +1,171 @@
+package yaml_test
```

Comment:
We have a very nice set of tests for comments in the node_test.go file. These need to be integrated there as well, and the ones unrelated to comments can go into the respective test files.

Reply:
I added most of the tests from this file (all single-document tests) to node_test.go as well.

Reply:
(02f3190efc7e74bda69720bda4cbb88eea340de9)